src/helper_lib/__init__.py

```diff
@@ -0,0 +1,178 @@
+# *******************************************************************************
+# Copyright (c) 2025 Contributors to the Eclipse Foundation
+#
+# See the NOTICE file(s) distributed with this work for additional
+# information regarding copyright ownership.
+#
+# This program and the accompanying materials are made available under the
+# terms of the Apache License Version 2.0 which is available at
+# https://www.apache.org/licenses/LICENSE-2.0
+#
+# SPDX-License-Identifier: Apache-2.0
+# *******************************************************************************
+
+import os
+import subprocess
+from pathlib import Path
+from sphinx_needs.logging import get_logger
+
+LOGGER = get_logger(__name__)
+
+
+def find_ws_root() -> Path | None:
+    """
+     Find the current MODULE.bazel file
+
+    Please note that this does behave differently depending on which command you use:
+    'bazel run' => Path
+    'bazel build' => None
+    'esbonio / direct execution' => None
+    """
+    ws_dir = os.environ.get("BUILD_WORKSPACE_DIRECTORY", None)
+    return Path(ws_dir) if ws_dir else None
+
+
+def find_git_root(start_path: str | Path = "") -> Path | None:
+    """
+    Find the git root directory starting from the given path or __file__.
+
+    Please note that this does behave differently depending on which command you use:
+    'bazel run' => Path
+    'bazel build' => None
+    'esbonio / direct execution' => Path (if executed without input)
+    """
+    if start_path == "":
+        start_path = __file__
+
+    git_root = Path(start_path).resolve()
+    # Note: recoursion_break is a variable that will stop
+    #       certain execution types, like esbonio
+    #       from recousing forever if it can't find a .git
+    #       in the 'home' or 'root' folder.
+    recoursion_break = False
+    while not (git_root / ".git").exists():
+        git_root = git_root.parent
+        if git_root == Path("/"):
+            # fallback to cwd when building with python -m sphinx docs _build -T
+            if recoursion_break:
+                return None
```

Comment:
Is it possible to write that loop somehow differently? I still don't get it. And there is no fallback to cwd?

Reply:
Consider [while + else](https://www.w3schools.com/python/gloss_python_while_else.asp). No idea whether that will make it better, but it sounds like a perfect fit.